qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+                "enable for position, amp error or abort": 1,
+                "enable for hw limit switch": 2,
+                "enable for all": 3,
+            },
+            docstring="enables or disables the motor to "
+            "automatically turn off when error occurs",
+        )
+
+        self.add_parameter(
+            "reverse_sw_limit",
+            unit="quadrature counts",
+            get_cmd=f"MG _BL{self._axis}",
+            get_parser=lambda s: int(float(s)),
+            set_cmd=self._set_reverse_sw_limit,
+            vals=Ints(-2147483648, 2147483647),
+            docstring="can be used to set software reverse limit for the motor."
+                      " motor motion will stop beyond this limit automatically."
+                      " default value is -2147483648. this value effectively "
+                      "disables the reverse limit.",
+        )
+
+        self.add_parameter(
+            "forward_sw_limit",
+            unit="quadrature counts",
+            get_cmd=f"MG _FL{self._axis}",
+            get_parser=lambda s: int(float(s)),
+            set_cmd=self._set_forward_sw_limit,
+            vals=Ints(-2147483648, 2147483647),
+            docstring="can be used to set software forward limit for the motor."
+                      " motor motion will stop beyond this limit automatically."
+                      " default value is 2147483647. this value effectively "
+                      "disables the forward limit.",
+        )
+
+    def _set_reverse_sw_limit(self, val: int) -> None:
+        """
+        Sets reverse software limit
+        """
+        self.write(f"BL{self._axis}={val}")
+
+    def _set_forward_sw_limit(self, val: int) -> None:
+        """
+        Sets forward software limit
+        """
+        self.write(f"FL{self._axis}={val}")
+
+    def _get_off_when_error_occurs(self) -> int:
+        """
+        Gets the status if motor is automatically set to turn off when error
+        occurs.
+        """
+        val = self.ask(f"MG _OE{self._axis}")
+
+        return int(val[0])
+
+    def _set_off_when_error_occurs(self, val: int) -> None:
+        """
+        sets the motor to turn off automatically when the error occurs
+        """
+        self.write(f"OE{self._axis}={val}")
+
+    def _set_deceleration(self, val: str) -> None:
+        """
+        set deceleration for the motor's motion
+        """
+        self.write(f"DC{self._axis}={val}")
+
+    def _set_acceleration(self, val: str) -> None:
+        """
+        set acceleration for the motor's motion
+        """
+        self.write(f"AC{self._axis}={val}")
+
+    def _set_speed(self, val: str) -> None:
+        """
+        sets speed for motor's motion
+        """
+        self.write(f"SP{self._axis}={val}")
+
+    def _set_relative_position(self, val: str) -> None:
+        """
+        sets relative position
+        """
+        self.write(f"PR{self._axis}={val}")
+
+    def off(self) -> None:
```

Comment:
should there be a corresponding "on"?

Reply:
servo_here is actually the "on" command. I have used the same terminology as in the manual for it. :)

Reply:
ah, ok, good :)